streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -221,6 +225,30 @@ public void shouldSendDataToTopicUsingProduced() {
         assertThat(processorSupplier.theCapturedProcessor().processed, equalTo(Collections.singletonList("e:f")));
     }
 
+    @Test
+    public void shouldSendDataToDynamicTopics() {
+        final StreamsBuilder builder = new StreamsBuilder();
+        final String input = "topic";
+        final KStream<String, String> stream = builder.stream(input, stringConsumed);
+        stream.to(new TopicNameExtractor<String, String>() {
+            @Override
+            public String extract(String key, String value, RecordContext recordContext) {
+                return recordContext.topic() + "-" + key + "-topic";
```

Comment:
Why not just use the key itself?

Reply:
I wanted to check that the record context fields and the key value are accessible in this test.

Reply:
Should be two tests instead? (Leave it up to you to decide -- don't insist on a change)